spec/rubocop/cli/auto_gen_config_spec.rb

```diff
@@ -951,33 +951,6 @@ def a; end
       YAML
     end
 
-    context 'when configurable parameter is an obsolete parameter' do
```

Comment:
This was a test added in <https://github.com/rubocop/rubocop/pull/10875>. However, this specification was incorrect. In fact, I think the cop_config was broken, causing duplicates to be displayed.

Reply:
Isn't it necessary to update the test as originally intended? I'm worried about losing a regression spec for #10875.

Reply:
Indeed it is. I will modify the test a bit and add it.

Reply:
Hmmm. I've tried running the following test and it seems to fail.
Maybe there is some other problem.
Should we include the resolution of that issue as part of this PR?

```ruby
    context 'when duplicated parameter' do
      it 'parameters are displayed without duplication' do
        create_file('.rubocop.yml', <<~YAML)
          AllCops:
            NewCops: enable
          Lint/AmbiguousBlockAssociation:
            AllowedMethods:
              - change
              - not_change
              - change
        YAML
        create_file('example1.rb', ['# frozen_string_literal: true', '', 'def fooBar; end'])
        create_file('example2.rb', ['# frozen_string_literal: true', '', 'def fooBar; end'])

        expect(cli.run(['--auto-gen-config'])).to eq(0)
        File.readlines('.rubocop_todo.yml')
        expect(File.readlines('.rubocop_todo.yml')[9..].join)
          .to eq(<<~YAML)
            # Configuration parameters: AllowedPatterns, IgnoredPatterns.
            # SupportedStyles: snake_case, camelCase
            # AllowedPatterns: change, not_change
            Naming/MethodName:
              EnforcedStyle: camelCase
          YAML
      end
    end
```

```
Failures:

  1) RuboCop::CLI --auto-gen-config --auto-gen-config when duplicated parameter parameters are displayed without duplication
     Failure/Error:
       expect(File.readlines('.rubocop_todo.yml')[9..].join)
         .to eq(<<~YAML)
           # Configuration parameters: AllowedPatterns, IgnoredPatterns.
           # SupportedStyles: snake_case, camelCase
           # AllowedPatterns: change, not_change
           Naming/MethodName:
             EnforcedStyle: camelCase
         YAML
     
       expected: "# Configuration parameters: AllowedPatterns, IgnoredPatterns.\n# SupportedStyles: snake_case, camelCase\n# AllowedPatterns: change, not_change\nNaming/MethodName:\n  EnforcedStyle: camelCase\n"
            got: "# Configuration parameters: AllowedPatterns, IgnoredPatterns.\n# SupportedStyles: snake_case, camelCase\nNaming/MethodName:\n  EnforcedStyle: camelCase\n"
     
       (compared using ==)
     
       Diff:
       @@ -1,6 +1,5 @@
        # Configuration parameters: AllowedPatterns, IgnoredPatterns.
        # SupportedStyles: snake_case, camelCase
       -# AllowedPatterns: change, not_change
        Naming/MethodName:
          EnforcedStyle: camelCase
       
     # ./spec/rubocop/cli/auto_gen_config_spec.rb:970:in `block (4 levels) in <top (required)>'
     # ./spec/support/cli_spec_behavior.rb:26:in `block (2 levels) in <top (required)>'
     # ./lib/rubocop/rspec/shared_contexts.rb:30:in `block (4 levels) in <top (required)>'
     # ./lib/rubocop/rspec/shared_contexts.rb:30:in `chdir'
     # ./lib/rubocop/rspec/shared_contexts.rb:30:in `block (3 levels) in <top (required)>'
     # ./lib/rubocop/rspec/shared_contexts.rb:7:in `block (2 levels) in <top (required)>'

Finished in 7.11 seconds (files took 1.06 seconds to load)
40 examples, 1 failure
```

Reply:
I see. I would like to run the same test code on the old version to check the existing behavior first.

Reply:
The configuration values like `# AllowedNames: at, by` that appear in `.rubocop_todo.yml` were not configuration values in `.rubocop.yml`, but default values in `config/defautl.yml`.
However, what appeared to be the `.rubocop.yml` configuration values in #10875 . This was due to the default values being corrupted by the problem fixed in this PR.
Therefore, it was necessary to duplicate parameter the `config/defautl.yml` here.